src/test/kotlin/com/atlassian/performance/tools/ssh/api/SshTest.kt

```diff
@@ -56,7 +56,8 @@ class SshTest {
         SshContainer().useSsh { sshHost ->
             installPing(sshHost)
 
-            val fail = sshHost.runInBackground("nonexistent-command")
+            val fail = sshHost.runInBackground("nonexistant-command")
+            sshHost.waitForAllProcessesToFinish("nonexistant-command", Duration.ofMillis(100))
             val failResult = fail.stop(Duration.ofMillis(20))
```

Comment:
So this fails due to`command.exitStatus must not be null`.
The `exitStatus` is supposed to be filled in the `if (exit-status)`, but it misses it and falls into `if (exit-signal)`:

![image](https://user-images.githubusercontent.com/1194704/189333795-c8340253-7d1f-4dd6-a591-c5d24b838513.png)

We can see that `exitSignal` = `INT`, so that's our `tryToInterrupt`. Where exactly is the race condition? :thinking:

Reply:
Lol, this happens when the `join` fails:
![image](https://user-images.githubusercontent.com/1194704/189335660-4cb2d56c-f922-4689-8fe8-77a41763c226.png)
So this is an error during error handling.

Reply:
The `close` in `readOutput` fails if the command was interrupted.
Now I'm getting real flakiness source:
> SSH command failed to finish in extended time (PT0.025S): SshjExecutedCommand(stdout=^Cbash: nonexistent-command-290: command not found

Reply:
BTW. It seems arbitrary to have both `SshResult` and `SshjExecutedCommand` and two ways of reading stdout

Reply:
I traced it down to ancient times behind the wall: https://bulldog.internal.atlassian.com/browse/JPT-292

Reply:
Hmm, I already had [the fixes](https://github.com/atlassian/ssh/compare/master...dagguh:ssh:fix-stop-flakiness?expand=1), but I was checking the effect of each commit via rebase (to populate the CHANGELOG correctly). And it turns out that  `failed to finish in extended time (PT0.025S)` happens even without the fixes. Sometimes it's a timeout, sometimes it's a null. I must have mixed up the observations around `exitStatus=null` with the wrong actual symptom. Gotta, really hammer down my understanding of the problem.